### ISSUE TITLE
Wrong dependency order when generating diffChangeLog for addColumn --> addPrimaryKey

### ISSUE THREAD

**Issue by szandany** [liquibase/liquibase#1722]
## Environment
**Liquibase Version**:
4.3.1
**Liquibase Integration & Version**: 
CLI
**Database Vendor & Version**:
PostgreSQL
**Operating System Type & Version**:
Linux, Windows
**changeLog format**:
XML, YAML, JSON, Formatted SQL

## Description
When generating a diffChangLog between the target schema and the reference schema, the resulting changeLog will reflect changes in the wrong order for addColumn --> addPrimaryKey change sets order.
This issue is happening for all changeLog formats.

## Steps To Reproduce
1. Run the following changeSet in the reference schema to create a table "person" --> add column --> add PK

```java
<changeSet  author="SteveZ"  id="createTable-person">
    <createTable tableName="person">  
        <column  name="address"  type="varchar(255)"/>  
    </createTable>  
</changeSet>
<changeSet id="SteveZ" author="234fre">
    <addColumn tableName="person">
        <column name="id" type="varchar(255)"/>
    </addColumn>
</changeSet>
<changeSet id="SteveZ" author="rtg45">
    <addPrimaryKey tableName="person" columnNames="id" constraintName="pk_person"/>
</changeSet>
```

1. Run the following changeSet in the target schema to create a table "person"

```java
<changeSet  author="SteveZ"  id="createTable-person">
    <createTable tableName="person">  
        <column  name="address"  type="varchar(255)"/>  
    </createTable>  
</changeSet>
```

1. Run `liquibase diffChangeLog` between the reference schema and the target schema.

## Actual Behavior
The resulting changeSets in the generated changeLog will look like the following:

```xml
<changeSet author="support.liquibase.net (generated)" id="1614270028483-1">
    <addPrimaryKey columnNames="id" constraintName="pk_person" tableName="person"/>
</changeSet>
<changeSet author="support.liquibase.net (generated)" id="1614270028483-2">
    <addColumn tableName="person">
        <column name="id" type="varchar(255 BYTE)">
            <constraints nullable="false"/>
        </column>
    </addColumn>
</changeSet>
```

The **primary key** creation and **add column** creation are in the wrong order.

## Expected/Desired Behavior
```
<changeSet author="support.liquibase.net (generated)" id="1614270028483-2">
    <addColumn tableName="person">
        <column name="id" type="varchar(255 BYTE)">
            <constraints nullable="false"/>
        </column>
    </addColumn>
</changeSet>
<changeSet author="support.liquibase.net (generated)" id="1614270028483-1">
    <addPrimaryKey columnNames="id" constraintName="pk_person" tableName="person"/>
</changeSet>
```

┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-1292) by [Unito](https://www.unito.io)
┆Fix Versions: Community 4.4.0,Liquibase 4.4.0

**Comment by wwillard7800** [comment]
@szandany Did you test this against any previous versions of Liquibase?

**Comment by szandany** [comment]
@wwillard7800 Yes.  This was tested with 4.0.0 as well.

**Comment by wwillard7800** [comment]
Did it work in that version?

**Comment by szandany** [comment]
No.  I got the same result.

**Comment by XDelphiGrl** [comment]
I'm going to give this a try now, @szandany.  This is a very basic use case that should be working (of course!). Thanks for bringing our attention to the issue.

**Comment by XDelphiGrl** [comment]
I can reproduce the bug following your steps, @szandany. I'd raise this to Product to make sure it gets added to a sprint. Thanks for the excellent bug report!

**Comment by nvoxland** [comment]
Fixed in #1758

**Comment by nvoxland** [comment]
Fix #1733 merged into master for 4.4.0

**Comment by nvoxland** [comment]
Reverted in 4.3.4 because it was causing issues. Will re-investigate a fix
